<a href="https://colab.research.google.com/github/Spitzanity/Sports-Analytics---RST407/blob/main/Exercise_2_Workbook.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

3.2:

In [1]:
# Given probabilities
P_A = 0.312  # Probability that home team scores at least 30 points
P_B = 0.219  # Probability that visiting team scores at least 30 points
P_A_and_B = 0.105  # Probability that both teams score at least 30 points

# Compute conditional probabilities
P_B_given_A = P_A_and_B / P_A
P_A_given_B = P_A_and_B / P_B

# Check for independence
P_A_times_P_B = P_A * P_B
independent = P_A_and_B == P_A_times_P_B

# Print results
print(f"P(B | A) = {P_B_given_A:.4f}")
print(f"P(A | B) = {P_A_given_B:.4f}")
print(f"Are A and B independent? {'Yes' if independent else 'No'}")

P(B | A) = 0.3365
P(A | B) = 0.4795
Are A and B independent? No


P(A∩B) is not equal to
P
(
A
)
P
(
B
)
P(A)P(B), so the events are not independent. This suggests that there is a relationship between these events—when one team scores at least 30 points, the other team is more likely to do so as well.

3.3:

In [2]:
# Given probabilities
P_H = 0.540  # Probability of home team win
P_T = 0.154  # Probability of game being tied after 5 innings
P_H_given_T = 0.149  # Probability of home team win given tie after 5 innings
P_V_given_T = 0.159  # Probability of visiting team win given tie after 5 innings

# Compare conditional probabilities
visiting_more_likely = P_V_given_T > P_H_given_T

# Print results
print(f"P(H | T) = {P_H_given_T:.4f}")
print(f"P(V | T) = {P_V_given_T:.4f}")
print(f"Is the visiting team more likely to win if tied after 5 innings? {'Yes' if visiting_more_likely else 'No'}")


P(H | T) = 0.1490
P(V | T) = 0.1590
Is the visiting team more likely to win if tied after 5 innings? Yes


Yes, based on the given conditional probabilities, the visiting team is slightly more likely to win when a game is tied after 5 innings. However, this does not necessarily mean the visiting team is generally more likely to win the game overall. The home team still wins more often in total, suggesting that home-field advantage plays a larger role over the full game.

3.5:

In [3]:
# Given probabilities
P_FG = 0.504  # Overall field goal percentage
P_FG_given_3P = 0.383  # Three-point shooting percentage
P_3P = 0.216  # Percentage of total shots that are three-pointers
P_2P = 1 - P_3P  # Percentage of total shots that are two-pointers

# Using the law of total probability to find two-point shooting percentage
P_FG_given_2P = (P_FG - (P_FG_given_3P * P_3P)) / P_2P

# Print results
print(f"P(FG | 2P) = {P_FG_given_2P:.4f} ({P_FG_given_2P * 100:.2f}%)")

P(FG | 2P) = 0.5373 (53.73%)


Thus, the Suns' two-point shooting percentage for the 2008-09 season was 53.73%.

3.6:

In [4]:
# Given probability
P_K = 0.35  # Probability that Max Scherzer strikes out a single batter

# Probability of striking out three consecutive batters
P_K3 = P_K ** 3

# Print result
print(f"P(K_3) = {P_K3:.4f} ({P_K3 * 100:.2f}%)")

P(K_3) = 0.0429 (4.29%)


Thus, the probability that Max Scherzer strikes out three consecutive batters is 0.0429 or 4.29%.

3.7:

In [5]:
import pandas as pd

# Load the dataset
file_path = "Dataset 3.2.xlsx"
xls = pd.ExcelFile(file_path)

# Load the data from the first sheet
df = pd.read_excel(xls, sheet_name='Sheet1')

# Trim whitespace from column names
df.columns = df.columns.str.strip()

# Define league-wide shot distribution
league_distribution = {
    "0-3": 0.281,
    "3-10": 0.156,
    "10-16": 0.106,
    "16-3pt": 0.120,
    "3pt": 0.337
}

# Compute adjusted shooting percentage for each player
adjusted_fg_percentages = {}

for index, row in df.iterrows():
    player_name = row["Player"]
    adjusted_fg = (
        (row["0-3"] * league_distribution["0-3"]) +
        (row["3-10"] * league_distribution["3-10"]) +
        (row["10-16"] * league_distribution["10-16"]) +
        (row["16-3pt"] * league_distribution["16-3pt"]) +
        (row["3pt"] * league_distribution["3pt"])
    )
    adjusted_fg_percentages[player_name] = adjusted_fg

# Convert results to a DataFrame for better display
adjusted_fg_df = pd.DataFrame(list(adjusted_fg_percentages.items()), columns=["Player", "Adjusted FG%"])

# Print results
print(adjusted_fg_df.to_string(index=False))


  Player  Adjusted FG%
  Durant       54.4821
   James       48.7039
  Korver       53.2564
Thompson       50.7052


Based on the adjustments, LeBron James is now actually in last. With Kevin Durant now in the lead.

3.8:

In [6]:
import pandas as pd

# Given values
p = 0.904  # Free throw success probability
n = 300  # Number of attempts for part (a)
N = 5000  # Total career attempts for part (b)

# Compute mean and standard deviation for part (a)
mean_X = n * p
std_X = (n * p * (1 - p)) ** 0.5

# Compute mean and standard deviation of proportion X/n
mean_X_n = mean_X / n
std_X_n = std_X / n

# Compute probability of making 97 consecutive free throws
p_97_consecutive = p ** 97

# Print results
results = f"""
(a) Mean of X: {mean_X:.1f}
    Standard deviation of X: {std_X:.1f}
    Mean of X/n: {mean_X_n:.3f}
    Standard deviation of X/n: {std_X_n:.3f}

(b) Probability of making 97 consecutive free throws: {p_97_consecutive:.10f}
"""

print(results)


(a) Mean of X: 271.2
    Standard deviation of X: 5.1
    Mean of X/n: 0.904
    Standard deviation of X/n: 0.017

(b) Probability of making 97 consecutive free throws: 0.0000560193



The probability of making 97 consecutive free throws is approximately 0.000056 (or 0.0056%), which is extremely low. This suggests that while Curry is an excellent free-throw shooter, breaking the record of 97 consecutive free throws would still be a rare event.

3.9:

In [10]:
import pandas as pd

# Load the dataset for FIP values
fip_file_path = "Dataset 3.3.xlsx"
fip_xls = pd.ExcelFile(fip_file_path)

# Load the data from the first sheet
fip_df = pd.read_excel(fip_xls, sheet_name='Sheet 1')

# Extract the five standout pitchers' FIP values
standout_fip = {
    "1946": (1.97, "Hal Newhouser"),
    "1968": (1.77, "Bob Gibson"),
    "1984": (1.69, "Dwight Gooden"),
    "1999": (1.39, "Pedro Martinez"),
    "2014": (1.81, "Clayton Kershaw")
}

# Compute mean and standard deviation for each year
z_scores = {}

for year, (fip_value, player_name) in standout_fip.items():
    column_name = f"Year {year}"
    mean_fip = fip_df[column_name].mean()
    std_fip = fip_df[column_name].std()

    # Compute Z-score
    z_score = (fip_value - mean_fip) / std_fip
    z_scores[year] = (z_score, player_name)

# Convert results to a DataFrame for better display
z_scores_df = pd.DataFrame([(year, data[0], data[1]) for year, data in z_scores.items()],
                            columns=["Year", "Z-Score", "Player"])

# Determine the most and least impressive performances
most_impressive_row = z_scores_df.loc[z_scores_df["Z-Score"].idxmin()]
least_impressive_row = z_scores_df.loc[z_scores_df["Z-Score"].idxmax()]

# Print results
print("Z-Scores for Standout Pitchers:")
print(z_scores_df.to_string(index=False))
print(f"\nMost Impressive Performance: {most_impressive_row['Player']} in {most_impressive_row['Year']} (Z = {most_impressive_row['Z-Score']:.2f})")
print(f"Least Impressive Performance: {least_impressive_row['Player']} in {least_impressive_row['Year']} (Z = {least_impressive_row['Z-Score']:.2f})")


Z-Scores for Standout Pitchers:
Year   Z-Score          Player
1946 -2.703083   Hal Newhouser
1968 -2.425673      Bob Gibson
1984 -3.817616   Dwight Gooden
1999 -4.366478  Pedro Martinez
2014 -3.003891 Clayton Kershaw

Most Impressive Performance: Pedro Martinez in 1999 (Z = -4.37)
Least Impressive Performance: Bob Gibson in 1968 (Z = -2.43)
